### PR TITLE
Guide episode new

### PR DESCRIPTION
**Pull request by tatitati**
[PROGRAMMES-6164](https://jira.dev.bbc.co.uk/browse/PROGRAMMES-6164)

---

- **Examples to test:**
http://sandbox.bbc.co.uk/programmes/b006q2x0/episodes/guide
http://sandbox.bbc.co.uk/programmes/b08n5s4l/episodes/guide.2013inc
http://sandbox.bbc.co.uk/programmes/b00rhg2r/episodes/guide
http://sandbox.bbc.co.uk/programmes/b006m8wd/episodes/guide

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: assets/js/episode-guide.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        var link = findCollapsableControlSeries(series),
+            panel = $('#' + link.attr('aria-controls')),
+            expandedDropdownFlag = link.find(classDropdownExpanded).eq(0),
+            collapsedDropdownFlag = link.find(classDropdownCollapsed).eq(0);
+
+        // expand series
+        series.attr('class', series.attr('data-activeclass'));
+
+        collapsedDropdownFlag.toggle();
+        expandedDropdownFlag.toggle();
```
Comment (tatitati) [inline assets/js/episode-guide.js]: This couple of ```toggles()``` work to display the collapse symbol ```>``` in the proper direction (```/\``` or ```>```). Before we were managing this with css, just changing a class which apply a ```transform:rotate(90deg)```. However now we add this icons with gelicons (SVG html tags), and add, remove classes seems not so easy for old jquery versions:
https://stackoverflow.com/questions/8638621/jquery-svg-why-cant-i-addclass

File: assets/js/episode-guide.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
+define(['jquery-1.9'], function ($) {
```
Comment (tatitati) [inline assets/js/episode-guide.js]: IMPORTANT NOTE: should I include istat already on this PR?

File: src/Ds2013/Presenters/Pages/EpisodesGuide/Collapsable/CollapsablePresenter.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
+<?php
+
+namespace App\Ds2013\Presenters\Pages\EpisodesGuide\Collapsable;
+
+use App\Ds2013\Presenter;
+use BBC\ProgrammesPagesService\Domain\Entity\ProgrammeContainer;
+
+class CollapsablePresenter extends Presenter
```
Comment (tatitati) [inline src/Ds2013/Presenters/Pages/EpisodesGuide/Collapsable/CollapsablePresenter.php]: QUESTION: 
not sure if we work in this way. I see that we don't create presenteres only for domain entities. Sometimes for Sub elements in our page. Is this ok?, should be better to use a twig partial?

File: templates/programme_episodes/guide_partials/listguide.html.twig [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+                    {% else %}
+                        {{ ds2013('collapsable', item, {}) }}
+                    {% endif %}
+                </li>
+            {% endfor %}
+        </ol>
+    {% else %}
+        <p class="note">{{ tr('episodes_none', 0)}}</p>
+    {% endif %}
+</div>
```
Comment (tatitati) [inline templates/programme_episodes/guide_partials/listguide.html.twig]: This partial allow to display Episodes and series (collapsables). If a nested series is found, this template is injected again, allowing to load more nested series and episodes again.

File: assets/js/rv-bootstrap.js [hunk 5]
```diff
@@ -1,4 +1,4 @@
-define(['jquery', 'lazyload', 'istats-tracking', 'respimg', 'lazysizes'],function($, Lazyload, IstatsTracking){
+define(['jquery', 'lazyload', 'istats-tracking', 'respimg', 'lazysizes', 'episode-guide'],function($, Lazyload, IstatsTracking){
```
Comment (votemike) [inline assets/js/rv-bootstrap.js]: Do we need this here? This will load the episode-guide JS on every page won't it?

File: src/Controller/ProgrammeEpisodes/GuideController.php [hunk 6]
```diff
@@ -18,6 +18,10 @@ public function __invoke(
         $page = $this->getPage();
         $limit = 30;
 
+        /**
+         * @var mixed[] $children
```
Comment (votemike) [inline src/Controller/ProgrammeEpisodes/GuideController.php]: Should series and episodes implement a common interface so that it can be type hinted on?

File: src/Ds2013/Presenters/Pages/EpisodesGuide/Collapsable/CollapsablePresenter.php [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    {
+        return $this->getParentContainer()->getTitle();
+    }
+
+    /**
+     * get MAIN PAGE specific series
+     */
+    public function getSeriesUrl() :string
+    {
+        return '/programmes/' . $this->getParentContainer()->getPid();
```
Comment (votemike) [inline src/Ds2013/Presenters/Pages/EpisodesGuide/Collapsable/CollapsablePresenter.php]: We shouldn't hard-code routes like this should we. We should use the router and a named route.

File: src/Ds2013/Presenters/Pages/EpisodesGuide/Collapsable/collapsable.html.twig [hunk 8]
```diff
@@ -0,0 +1,40 @@
+<div typeof="Season"
+     resource="/programmes/{{ collapsable.getPid() }}"
```
Comment (votemike) [inline src/Ds2013/Presenters/Pages/EpisodesGuide/Collapsable/collapsable.html.twig]: Use the router. Don't hardcode routes.

File: assets/js/episode-guide.js [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    /**
+     * "link series" work to load and display the collapsed content (episodes) of a series.
+     */
+    function loadEpisodesFromSelectedSeries(series, redirectIfFail) {
+        var seriesLink = findCollapsableControlSeries(series),
+            href = seriesLink.attr('href');
+
+        series.addClass('js-inprogress');
```
Comment (votemike) [inline assets/js/episode-guide.js]: Does this need removing at some point?

File: src/Controller/ProgrammeEpisodes/GuideController.php (hunk 6 above)
Comment (tatitati) [reply src/Controller/ProgrammeEpisodes/GuideController.php]: interesting....

Comment (felipeparaujo) [reply src/Controller/ProgrammeEpisodes/GuideController.php]: A new interface seems unecessary. It would complicate an even more complicated data layer. Just use CoreEntity here, should be enough

Comment (votemike) [reply src/Controller/ProgrammeEpisodes/GuideController.php]: Not `Programme`?

Comment (felipeparaujo) [reply src/Controller/ProgrammeEpisodes/GuideController.php]: Yes, `Programme`. Not `CoreEntity`

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+                     class="js-guideitem highlight-box--list br-keyline br-blocklink-page br-page-linkhover-onbg015--hover"
+                     data-inactiveclass="js-guideitem highlight-box--list br-keyline br-blocklink-page br-page-linkhover-onbg015--hover"
+                     data-activeclass="js-guideitem br-box-subtle island js-active"
+                     data-fetched="false"
+                     data-incpath="{{ path('programme_episodes_guide_partial', {'pid': programme.getPid()}) }}">
+
+                    <span class="loading-spinner pull--right-spaced hidden js-inprogress-showme"></span>
+
+                    {#Title of this season#}
+                    <h2 class="gamma no-margin">
```
Comment (felipeparaujo) [inline src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: These headings here should be parametrized, as well as the ones we pass to the episode object. They should increase according to the nesting level of the series/episode.

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+            {%- else -%}
+                {#EPISODE#}
+                <div class="js-guideitem highlight-box--list block-link block-link--steal br-keyline br-blocklink-page br-page-linkhover-onbg015--hover">
+                    <div class="grid-wrapper">
+                        <div class="grid 2/3@bpw2 2/3@bpe">
+                            {{ ds2013('programme', programme, {
+                                'context_programme': episode_guide_list.getContextProgramme(),
+                                title_options: {
+                                    'title_format': 'item::ancestry',
+                                    'title_tag': 'h3',
```
Comment (felipeparaujo) [inline src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: read comment up there ^

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+                                    'title_classes': 'gamma'
+                                }
+                            }) }}
+                        </div>
+                        <div class="grid 1/3@bpw2 1/3@bpe hidden grid-visible@bpw2 grid-visible@bpe">
+                            <div class="grid-wrapper">
+                                <div class="grid 1/2">
+                                    {%- if episode_guide_list.getUpcomingBroadcastsFromProgramme(programme.getPid()) is not null -%}
+                                        <h3 class="delta no-margin">{{ tr('next_on') }}</h3>
```
Comment (felipeparaujo) [inline src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: same thing as the other heading tags

File: tests/Controller/ProgrammeEpisodes/DomParsers/EpisodesGuideDomParser.php [hunk 13]
```diff
@@ -0,0 +1,60 @@
+<?php
```
Comment (felipeparaujo) [inline tests/Controller/ProgrammeEpisodes/DomParsers/EpisodesGuideDomParser.php]: This whole class is a bit of an overkill...

File: src/Controller/ProgrammeEpisodes/GuidePartialController.php [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+        return $this->renderWithChrome('programme_episodes/guide.2013inc.html.twig', [
+            'programme' => $programme,
+            'children' => $children,
+            'totalChildrenCount' => $totalChildrenCount,
+            'upcomingBroadcasts' => $upcomingBroadcasts,
+        ]);
+    }
+
+    public function getUpcomingBroadcastsIndexedByProgrammePid(
```
Comment (BjornTwachtmann) [inline src/Controller/ProgrammeEpisodes/GuidePartialController.php]: This shouldn't be duplicated. It also shouldn't be public.

File: src/Controller/ProgrammeEpisodes/GuidePartialController.php [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+use BBC\ProgrammesPagesService\Domain\Entity\ProgrammeContainer;
+use BBC\ProgrammesPagesService\Service\CollapsedBroadcastsService;
+use BBC\ProgrammesPagesService\Service\ProgrammesAggregationService;
+use BBC\ProgrammesPagesService\Service\ProgrammesService;
+
+class GuidePartialController extends BaseController
+{
+    public function __invoke(
+        ProgrammeContainer $programme,
+        PresenterFactory $presenterFactory,
```
Comment (BjornTwachtmann) [inline src/Controller/ProgrammeEpisodes/GuidePartialController.php]: Some of your parameters here are unused

File: src/Controller/ProgrammeEpisodes/GuidePartialController.php [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+
+        $children = $programmesService->findEpisodeGuideChildren($programme, $limit);
+        if (!$children) {
+            throw $this->createNotFoundException('Page does not exist');
+        }
+
+        $totalChildrenCount = $programmesService->countEpisodeGuideChildren($programme);
+        $upcomingBroadcasts = $this->getUpcomingBroadcastsIndexedByProgrammePid($programme, $collapsedBroadcastService);
+
+        $this->setIstatsExtraLabels([
```
Comment (BjornTwachtmann) [inline src/Controller/ProgrammeEpisodes/GuidePartialController.php]: Please check, but I think this probably does nothing.

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+                    {#Title of this season#}
+                    <h2 class="gamma no-margin">
+                        <a property="url"
+                           resource="{{ path('find_by_pid', {'pid': programme.getPid()}) }}"
+                           class="js-serieslink istats-seriesopen block-link link-complex"
+                           href="{{ path('programme_episodes_guide_partial', {'pid': programme.getPid()}) }}"
+                           aria-expanded="false"
+                           aria-controls="series-{{ programme.getPid() }}">
+
+                            <button class="js-dropdown js-dropdown--collapsed">{{ gelicon('core', 'next', 'gelicon--delta gelicon--trailing') }}</button>
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: Please remove <button> and use the relevant aria-role on the anchor tag

File: templates/base_ds2013.html.twig [hunk 18]
```diff
@@ -72,9 +72,10 @@
                     'jquery-1.9': "https://static.bbci.co.uk/frameworks/jquery/0.4.1/sharedmodules/jquery-1.9.1",
                     'rv-bootstrap': "{{ asset_js('js/rv-bootstrap.js') }}",
                     'timezone-notification': "{{ asset_js('js/timezone-notification.js') }}",
-                    'comscorews': "{{ asset_js('js/comscorews.js') }}"
+                    'comscorews': "{{ asset_js('js/comscorews.js') }}",
+                    'episode-guide': "{{ asset_js('js/episode-guide.js') }}"
                 },
-                "priority": ['jquery-1.9', 'rv-bootstrap', 'timezone-notification'],
+                "priority": ['jquery-1.9', 'rv-bootstrap', 'timezone-notification', 'episode-guide'],
```
Comment (BjornTwachtmann) [inline templates/base_ds2013.html.twig]: Why is this here?

File: assets/js/episode-guide.js [hunk 19]
```diff
@@ -0,0 +1,179 @@
...
+        /**
+         * One series can be collapsed (icon: >) or expanded (icon: V)
+         *
+         * @Example 'V SEASON 1'
+         */
+        var showExpandedIcon = function(seriesLinkClicked)
+        {
+            var classesExpanded = seriesLinkClicked.find('[aria-role="button"]').eq(0).attr('data-expandedclass');
+            var gelicon = seriesLinkClicked.find(geliconClass).eq(0);
+            gelicon.attr('class', classesExpanded);
```
Comment (tatitati) [inline assets/js/episode-guide.js]: How to add/remove class to svg elements with jquery.
https://martinwolf.org/blog/2014/12/adding-and-removing-classes-from-svg-elements-with-jquery

File: src/Controller/ProgrammeEpisodes/GuidePartialController.php [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+declare(strict_types = 1);
+namespace App\Controller\ProgrammeEpisodes;
+
+use App\Ds2013\PresenterFactory;
+use BBC\ProgrammesPagesService\Domain\Entity\ProgrammeContainer;
+use BBC\ProgrammesPagesService\Service\CollapsedBroadcastsService;
+use BBC\ProgrammesPagesService\Service\ProgrammesAggregationService;
+use BBC\ProgrammesPagesService\Service\ProgrammesService;
+
+class GuidePartialController extends GuideController
```
Comment (tatitati) [inline src/Controller/ProgrammeEpisodes/GuidePartialController.php]: To avoid some duplication, this is extending the controller above

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/EpisodeGuideListPresenter.php [hunk 21]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    public function getQueryParamForNextNestedLevel() :string
+    {
+        return "nestedlevel=" . ($this->nestedLevel + 1);
+    }
+
+    public function getHeadingTag(): string
+    {
+        return 'H' . $this->nestedLevel;
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Domain/EpisodeGuideList/EpisodeGuideListPresenter.php]: Shouldn't be a capital letter

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig [hunk 22]
```diff
@@ -0,0 +1,76 @@
...
+            {%- else -%}
+                {#EPISODE#}
+                <div class="js-guideitem highlight-box--list block-link block-link--steal br-keyline br-blocklink-page br-page-linkhover-onbg015--hover">
+                    <div class="grid-wrapper">
+                        <div class="grid 2/3@bpw2 2/3@bpe">
+                            {{ ds2013('programme', programme, {
+                                'context_programme': episode_guide_list.getContextProgramme(),
+                                title_options: {
+                                    'title_format': 'item::ancestry',
+                                    'title_tag': episode_guide_list.getHeadingTag(),
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: Do you want the programme object to have the same heading level as the parent?

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig [hunk 23]
```diff
@@ -0,0 +1,76 @@
...
+                    <{{ episode_guide_list.getHeadingTag() }} class="gamma no-margin">
+                        <a property="url"
+                           resource="{{ path('find_by_pid', {'pid': programme.getPid()}) }}"
+                           class="js-serieslink istats-seriesopen block-link link-complex"
+                           href="{{ path('programme_episodes_guide_partial', {'pid': programme.getPid()}) }}"
+                           aria-expanded="false"
+                           aria-controls="series-{{ programme.getPid() }}">
+
+                            <span aria-role="button"
+                                  data-expandedclass="gelicon gelicon--delta gelicon--trailing ml__button--dropdown ml__button--dropdown-expanded"
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: The amount of data attributes and classes in them feels like overkill. 
Can't you just make the JS add/remove/toggle a single class here, rather than listing all classes 3 times?

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig (hunk 22 above)
Comment (tatitati) [reply src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: well, actually is not the case:
I would have something like:
```
<h2>Series
        <h3>Series
        <h3>Episode
        <h3>Series
<h2>Episode
......
```
So everything in the same leven keep the same header

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig (hunk 23 above)
Comment (tatitati) [reply src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: https://martinwolf.org/blog/2014/12/adding-and-removing-classes-from-svg-elements-with-jquery

File: src/Controller/ProgrammeEpisodes/GuidePartialController.php [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+{
+    public function __invoke(
+        ProgrammeContainer $programme,
+        PresenterFactory $presenterFactory,
+        ProgrammesService $programmesService,
+        ProgrammesAggregationService $programmeAggregationService,
+        CollapsedBroadcastsService $collapsedBroadcastService
+    ) {
+
+        $nestedLevel = $this->request()->query->get('nestedlevel');
```
Comment (votemike) [inline src/Controller/ProgrammeEpisodes/GuidePartialController.php]: The request is passed through isn't it. Don't we usually get the parameters from that version of the request?

File: src/Ds2013/Css/Components/_dropdown.scss [hunk 25]
```diff
@@ -0,0 +1,14 @@
+.ml__button--dropdown {
+    margin-left: -$basic-spacing;
+    top: $basic-half-spacing / 2;
+}
+
+.ml__button--dropdown-expanded {
+    transform: rotate(90deg);
+    transition: transform 0.25s linear;
```
Comment (votemike) [inline src/Ds2013/Css/Components/_dropdown.scss]: Would the transition be better on `.ml__button--dropdown`?

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/EpisodeGuideListPresenter.php [hunk 26]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        return $this->programmes;
+    }
+
+    public function getContextProgramme(): ProgrammeContainer
+    {
+        return $this->contextProgramme;
+    }
+
+    public function getUpcomingBroadcastsFromProgramme(Pid $pid)
```
Comment (votemike) [inline src/Ds2013/Presenters/Domain/EpisodeGuideList/EpisodeGuideListPresenter.php]: What does this return?

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/EpisodeGuideListPresenter.php [hunk 27]
```diff
@@ -0,0 +1,78 @@
...
+use BBC\ProgrammesPagesService\Domain\Entity\Programme;
+use BBC\ProgrammesPagesService\Domain\Entity\ProgrammeContainer;
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+
+class EpisodeGuideListPresenter extends Presenter
+{
+    /** @var ProgrammeContainer */
+    private $contextProgramme;
+
+    /** @var Programme[] */
```
Comment (votemike) [inline src/Ds2013/Presenters/Domain/EpisodeGuideList/EpisodeGuideListPresenter.php]: These can be *any* type of Programme? Brand, Clip etc...?

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig [hunk 28]
```diff
@@ -0,0 +1,76 @@
...
+                     data-inactiveclass="js-guideitem highlight-box--list br-keyline br-blocklink-page br-page-linkhover-onbg015--hover"
+                     data-activeclass="js-guideitem br-box-subtle island js-active"
+                     data-fetched="false"
+                     data-incpath="{{ path('programme_episodes_guide_partial', {'pid': programme.getPid() }) }}?{{ episode_guide_list.getQueryParamForNextNestedLevel() }}">
+
+                    <span class="loading-spinner pull--right-spaced hidden js-inprogress-showme"></span>
+
+                    {#Title of this season#}
+                    <{{ episode_guide_list.getHeadingTag() }} class="gamma no-margin">
+                        <a property="url"
```
Comment (votemike) [inline src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: This is just a toggle isn't it? If so, it should be a button, not a link.

File: templates/programme_episodes/guide.2013inc.html.twig [hunk 29]
```diff
@@ -1,3 +1,14 @@
...
+{% block body %}
+<h{{ nestedlevel }} class="visually-hidden">Episodes</h{{ nestedlevel }}>
+
+<div>
+    {{ ds2013('episodeGuideList', programme, children, upcomingBroadcasts, nestedlevel) }}
 </div>
+
+{%- if totalChildrenCount > 30 -%}
+    <p class="no-margin">
+        <a class="text--center block-link delta island--squashed br-linkinvert" href="{{ path('programme_episodes_guide', {'pid': programme.getPid()}) }}?page=2">More episodes</a>
```
Comment (votemike) [inline templates/programme_episodes/guide.2013inc.html.twig]: Will it never be page 3?

File: assets/js/episode-guide.js [hunk 30]
```diff
@@ -0,0 +1,178 @@
...
+        };
+
+        var isSeriesContentAlreadyLoaded = function(series)
+        {
+            return series.attr('data-fetched') == 'true';
+        };
+
+        var isSeriesContentLoadingNow = function(series)
+        {
+            return series.hasClass('js-inprogress');
```
Comment (votemike) [inline assets/js/episode-guide.js]: Why is the Javascript logic based on classes being added and removed from the DOM? Why is the status not stored as part of an object in the JS?

File: assets/js/episode-guide.js [hunk 31]
```diff
@@ -0,0 +1,178 @@
...
+        };
+
+        var openSeriesIfNeeded = function()
```
Comment (votemike) [inline assets/js/episode-guide.js]: This function name doesn't quite make sense.

File: assets/js/episode-guide.js [hunk 32]
```diff
@@ -0,0 +1,178 @@
...
+        {
+            // special requirement: in case that the first element is a
+            // series, we open it automatically
+            var firstItem = $(classAnyItemInGuide).eq(0);
+            if (firstItem.length > 0 && firstItem.is('[data-incpath]')) {
+                firstItem.find(classDropdownLink).eq(0).trigger('click');
```
Comment (votemike) [inline assets/js/episode-guide.js]: Why are you triggering clicks? Why not just call the method that is triggered by the click?

File: assets/js/episode-guide.js [hunk 33]
```diff
@@ -0,0 +1,178 @@
...
+        };
+
+        /**
+         * One series can be collapsed (icon: >) or expanded (icon: V)
+         *
+         * @Example 'V SEASON 1'
+         */
+        var showExpandedIcon = function(seriesLinkClicked)
+        {
+            var classesExpanded = seriesLinkClicked.find('[aria-role="button"]').eq(0).attr('data-expandedclass');
```
Comment (votemike) [inline assets/js/episode-guide.js]: Why are you finding things based on a[aria-role="button"]a? `aria-role` is for accessibility, **not** for selecting elements.

File: assets/js/episode-guide.js [hunk 34]
```diff
@@ -0,0 +1,178 @@
...
+        };
+
+        /**
+         * We want to show the content of a selected series, so initially we have to load
+         * the content of it to later display it. In case that the content is already loaded,
+         * we go through the happy way (just display it)
+         */
+        var expandSeriesSelected = function(series) {
+            var seriesLinkClicked = series.find(classDropdownLink),
+                panel = $('#' + seriesLinkClicked.attr('aria-controls'));
```
Comment (votemike) [inline assets/js/episode-guide.js]: There should not be any mention of aria in this file.

File: assets/js/episode-guide.js [hunk 35]
```diff
@@ -0,0 +1,178 @@
...
+        /**
+         * Load content (list of episodes or nested series) for the selected series.
+         */
+        var loadEpisodesFromSelectedSeries = function(series, shouldRedirectIfFail) {
+            var url=series.attr('data-incpath');
+
+            $.ajax({
+                url: url,
+                beforeSend: function() {
+                    series.addClass('js-inprogress');
```
Comment (votemike) [inline assets/js/episode-guide.js]: Why set a class? Just store the fact the the ajax is in progress in the JS.

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/EpisodeGuideListPresenter.php (hunk 27 above)
Comment (tatitati) [reply src/Ds2013/Presenters/Domain/EpisodeGuideList/EpisodeGuideListPresenter.php]: They can be Episodes and Series only.

File: templates/programme_episodes/guide.2013inc.html.twig (hunk 29 above)
Comment (tatitati) [reply templates/programme_episodes/guide.2013inc.html.twig]: Yes, explained with a use case:
We have a brand with 120 series inside. Because this is too much to display in once we display only 30 and a paginator.

Now other case (your question):
One of these series has 200 episodes inside. In this case we display only 30, and after a link at the bottom displaying "More episodes". If clicked, it takes you to the second page of that series because we display only the first one (the 30 I said)

File: assets/js/episode-guide.js (hunk 35 above)
Comment (tatitati) [reply assets/js/episode-guide.js]: we set the class to display the spinner loading

File: src/Controller/ProgrammeEpisodes/GuidePartialController.php (hunk 24 above)
Comment (tatitati) [reply src/Controller/ProgrammeEpisodes/GuidePartialController.php]: Yeah. usually they are embeded in the url like /{pid}/. 
In this case I thought that this nested level doesn't affect to the data you receive. It just change the headers such as:
```
?nestedlevel=1 ---> <h1>
?nestedlevel=2 ---> <h2>
....
```
I thought that because is more related with tweaks in the template and can be even optional might be better to add just a query param

File: templates/programme_episodes/guide.2013inc.html.twig (hunk 29 above)
Comment (votemike) [reply templates/programme_episodes/guide.2013inc.html.twig]: Ahh, because on page 1 of that other page, it also displays the same 30.
If we are using this 30 in multiple places, should we make it a constant?

File: src/Controller/ProgrammeEpisodes/GuidePartialController.php (hunk 24 above)
Comment (votemike) [reply src/Controller/ProgrammeEpisodes/GuidePartialController.php]: I mean coming through the `__invoke` method with `Request $request,` and then access it 
`$request->query->get('nestedlevel');`

Comment (tatitati) [reply src/Controller/ProgrammeEpisodes/GuidePartialController.php]: 👍

File: assets/js/episode-guide.js (hunk 30 above)
Comment (tatitati) [reply assets/js/episode-guide.js]: We don't make only one call ajax. The user can select quickly three or 20 series that will be opened when finished each request. It was also done in that way in episodes-guide.js on V2

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig (hunk 28 above)
Comment (tatitati) [reply src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: Yesterday we were discussing this @BjornTwachtmann and @felipeparaujo , we decided to keep it as a link and instead add the aria-role="button"

Comment (votemike) [reply src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: Tab to it. Press enter.

Comment (votemike) [reply src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: Turns out it breaks even if you just click on it....

Comment (tatitati) [reply src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: Maybe yarn run gulp?

File: templates/programme_episodes/guide.2013inc.html.twig (hunk 29 above)
Comment (tatitati) [reply templates/programme_episodes/guide.2013inc.html.twig]: yeah , I added it

File: tests/Controller/ProgrammeEpisodes/DomParsers/EpisodesGuideDomParser.php (hunk 13 above)
Comment (tatitati) [reply tests/Controller/ProgrammeEpisodes/DomParsers/EpisodesGuideDomParser.php]: I really think that it helps to have more expressive tests. 
```php
$this->assertEquals(3, $episodesGuideDom->findGuideItemsInDom()->count());
$this->assertEquals(2, $episodesGuideDom->findSeriesInDom()->count());
$this->assertEquals(1, $episodesGuideDom->findEpisodesInDom()->count());
```
vs
```php
$this->assertEquals(3, $crawler->filter('.js-guideitem')->count());
$this->assertEquals(2, $crawler->filter('[typeof="Season"]')->count());
$this->assertEquals(1, $crawler->filter('.js-guideitem .programme--episode')->count());
```
Instead of a bunch of selectors that don't give you too much information you can really understand what is being parsed and create more complex selectors. For instance one interesting is this:

```php
$this->assertFalse($episodesGuideDom->isPartialTemplate());
```

vs:
```php
// what doesnt mean that there is no footer??
$this->assertFalse($crawler->filter('.footer')->count() === 0);
```

File: tests/DataFixtures/ORM/ProgrammeEpisodes/ProgrammeItemFixture.php [hunk 36]
```diff
@@ -4,21 +4,77 @@
...
+        // standalone episodes (not nested in any series)
+        $this->buildEpisode('p3000000', 'B1-E1', $brand);
+
+        // Series with simple nested episodes
+        $this->buildEpisode('p3000001', 'B1-S1-E1', $series1);
+        $this->buildEpisode('p3000002', 'B1-S1-E2', $series1);
+
+        // Series with more nested series
+        $this->buildEpisode('p3000003', 'B1-S2-S1-E1', $nestedSeries);
+        $this->buildEpisode('p3000004', 'B1-S2-S1-E2', $nestedSeries);
```
Comment (tatitati) [inline tests/DataFixtures/ORM/ProgrammeEpisodes/ProgrammeItemFixture.php]: We can use Composite pattern to create this hierarchies better, something like
```php
$brand->addChild(
         $serie->addChild(
                    $episode
         )
)->flush()
```

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig [hunk 37]
```diff
@@ -0,0 +1,77 @@
...
+                           resource="{{ path('find_by_pid', {'pid': programme.getPid()}) }}"
+                           class="js-serieslink istats-seriesopen block-link link-complex"
+                           href="{{ path('programme_episodes_guide_partial', {'pid': programme.getPid()}) }}"
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: The href needs to go to the actual episode page. We need another attribute for the lazy load.

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig (hunk 28 above)
Comment (votemike) [reply src/Ds2013/Presenters/Domain/EpisodeGuideList/episode_guide_list.html.twig]: In any case. If you don't leave the page, this shouldn't be a link.

File: tests/Controller/ProgrammeEpisodes/DomParsers/EpisodesGuideDomParser.php (hunk 13 above)
Comment (felipeparaujo) [reply tests/Controller/ProgrammeEpisodes/DomParsers/EpisodesGuideDomParser.php]: The problem I see here is that we'll end up with all of these different ideas that could be useful, but are implemented in only a couple of places. This will make the code harder to read in the future, and we'll wonder why this one bit of code is different. 

I'm not saying this is bad, but we need to have standards and stick to them. If we want to do thing this way, fine, but we'll need to update the other tests to follow this standard (in another PR)

File: src/Ds2013/Css/Components/_dropdown.scss [hunk 38]
```diff
@@ -0,0 +1,20 @@
...
+    position: relative;
+    top: $basic-half-spacing / 2;
+    transition: transform 0.25s linear;
+}
+
+.dropdown--icon--expanded {
+    transform: rotate(90deg);
+}
+
+
```
Comment (felipeparaujo) [inline src/Ds2013/Css/Components/_dropdown.scss]: Got a bit of extra whitespace here

File: tests/Controller/ProgrammeEpisodes/DomParsers/EpisodesGuideDomParser.php (hunk 13 above)
Comment (tatitati) [reply tests/Controller/ProgrammeEpisodes/DomParsers/EpisodesGuideDomParser.php]: aaah, ok, I get your point. OK, I will do as we have already.

File: src/Ds2013/Presenters/Domain/EpisodeGuideList/EpisodeGuideListPresenter.php [hunk 39]
```diff
@@ -0,0 +1,65 @@
+<?php
+declare(strict_types=1);
+
+namespace App\Ds2013\Presenters\Domain\EpisodeGuideList;
```
Comment (BjornTwachtmann) [inline src/Ds2013/Presenters/Domain/EpisodeGuideList/EpisodeGuideListPresenter.php]: This is not a domain object
